isaactest/tests/board_builder.py

```diff
@@ -0,0 +1,163 @@
+
+        subject_field.select_by_visible_text('Physics')
+        time.sleep(WAIT_DUR)
+        check_all_results(driver,
+                          lambda question_result: 'physics' in question_result.text.lower(),
+                          lambda question_result_text: '"physics" was not found in "{}"'.format(question_result_text.lower()))
+
+        log(INFO, "Test level field filtering")
+        level_field = Select(driver.find_element_by_xpath('//select[@ng-model="questionSearchLevel"]'))
+        level_field.select_by_visible_text('2')
+        time.sleep(WAIT_DUR)
+        check_all_results(driver,
+                          lambda question_result: 'level 2' in question_result.text.lower(),
+                          lambda question_result_text: '"level 2" was not found in "{}"'.format(question_result_text.lower()))
+
+        log(INFO, "Test query field filtering")
+        query_field = driver.find_element_by_xpath('//input[@ng-model="questionSearchText"]')
+        query_field.send_keys('toboggan')
+        time.sleep(WAIT_DUR)
+        check_all_results(driver,
+                          lambda question_result: 'toboggan' in question_result.text.lower(),
+                          lambda question_result_text: '"toboggan" was not found in "{}"'.format(question_result_text.lower()))
+
+        log(INFO, "Check book links work")
+        mastering_physics_link = driver.find_element_by_xpath('//a[text() = "Mastering Physics"]')
+        mastering_physics_link.click()
+        time.sleep(WAIT_DUR)
+        check_all_results(driver,
+                          lambda question_result: all(keyword in question_result.text.lower() for keyword in ['book', 'physics', 'physics_skills_14']),
+                          lambda question_result_text: '"book", "physics" or "physics_skills_14" was not found in "{}"'.format(question_result_text.lower()))
+
+        mastering_chemistry_link = driver.find_element_by_xpath('//a[text() = "Mastering Chemistry"]')
+        mastering_chemistry_link.click()
+        time.sleep(WAIT_DUR)
+        check_all_results(driver,
+                          lambda question_result: all(keyword in question_result.text.lower() for keyword in ['chemistry', 'chemistry_16']),
+                          lambda question_result_text: '"chemistry" or "chemistry_16" was not found in "{}"'.format(question_result_text.lower()))
+
+        log(INFO, "Build a board")
+        question_checkboxes = driver.find_elements_by_xpath("//ul[@class='no-bullet results-list ng-scope']//li//input[@type='checkbox' and @ng-model='enabledQuestions[question.id]']")
+        for i in xrange(10):
+            element = question_checkboxes[i]
+            driver.execute_script("arguments[0].scrollIntoView(true);", element)
+            element.click()
+        time.sleep(1)
+        assert not len(driver.find_elements_by_xpath('//h4[text() = "Too Many Questions"]')), 'Error displayed when not expected'
```

Comment:
Using a naive `time.sleep(1)` may or may not catch the pop-up, depending on how slow the JavaScript is etc.